integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/http2/EchoHttp2ServerHandler.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.mockbackend.http2;
+
+import io.netty.buffer.EmptyByteBuf;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http2.DefaultHttp2DataFrame;
+import io.netty.handler.codec.http2.DefaultHttp2Headers;
+import io.netty.handler.codec.http2.DefaultHttp2HeadersFrame;
+import io.netty.handler.codec.http2.DefaultHttp2WindowUpdateFrame;
+import io.netty.handler.codec.http2.Http2DataFrame;
+import io.netty.handler.codec.http2.Http2FrameStream;
+import io.netty.handler.codec.http2.Http2Headers;
+import io.netty.handler.codec.http2.Http2HeadersFrame;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+
+/**
+ * Handler implementation for the http/2 echo server without content
+ * aggregation. This echo backs the header/data
+ * frames as soon as they arrive without any content aggregation against stream
+ * id.
+ */
+public class EchoHttp2ServerHandler extends ChannelDuplexHandler {
```

Comment:
I think we need to check for the case where trailers are received by the backend and sent back to the client. (In that case, the endstream flag is sent in the trailers not the dataframe)

Reply:
As [VirajSalaka](https://github.com/VirajSalaka) and I checked, there wasn't a way to send/read trailers with http2 client and  we are not going to test this scenario at this time